components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAOImpl.java

```diff
@@ -0,0 +1,1099 @@
+            try (NamedPreparedStatement statement = new NamedPreparedStatement(connection, ADD_SCIM_ROLE_ID_SQL)) {
+                statement.setInt(RoleTableColumns.TENANT_ID, tenantId);
+                statement.setString(RoleTableColumns.ROLE_NAME, roleName);
+                statement.setString(RoleTableColumns.ATTR_NAME, RoleConstants.ID_URI);
+                statement.setString(RoleTableColumns.ATTR_VALUE, id);
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            String errorMessage = "Error while adding the the roleID: %s for the role: %s in the tenantDomain: %s";
+            throw new IdentityRoleManagementServerException(UNEXPECTED_SERVER_ERROR.getCode(),
+                    String.format(errorMessage, id, roleName, tenantDomain), e);
+        }
+        return id;
+    }
+
+    @Override
+    public Role getRole(String roleID, String tenantDomain) throws IdentityRoleManagementException {
+
+        Role role = new Role();
+        String roleName = getRoleNameByID(roleID, tenantDomain);
+        role.setId(roleID);
+        role.setName(roleName);
+        role.setTenantDomain(tenantDomain);
+        role.setUsers(getUserListOfRole(roleName, tenantDomain));
+        role.setGroups(getGroupListOfRole(roleName, tenantDomain));
+        return role;
+    }
+
+    @Override
+    public List<RoleBasicInfo> getRoles(int limit, int offset, String sortBy, String sortOrder, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        limit = validateLimit(limit);
+        offset = validateOffset(offset);
+        validateAttributesForSorting(sortBy, sortOrder);
+        List<RoleBasicInfo> roles;
+        try (Connection connection = IdentityDatabaseUtil.getUserDBConnection()) {
+            String databaseProductName = connection.getMetaData().getDatabaseProductName();
+            try (NamedPreparedStatement statement = new NamedPreparedStatement(connection,
+                    getDBTypeSpecificRolesRetrievalQuery(databaseProductName))) {
+                statement.setInt(RoleTableColumns.UM_TENANT_ID, tenantId);
+                roles = processListRolesQuery(limit, offset, statement, tenantDomain);
+            }
+        } catch (SQLException e) {
+            throw new IdentityRoleManagementServerException(UNEXPECTED_SERVER_ERROR.getCode(),
+                    "Error while listing roles in tenantDomain: " + tenantDomain, e);
+        }
+        return Collections.unmodifiableList(roles);
+    }
+
+    @Override
+    public List<RoleBasicInfo> getRoles(String filter, int limit, int offset, String sortBy, String sortOrder,
+            String tenantDomain) throws IdentityRoleManagementException {
+
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        if (StringUtils.isBlank(filter) || "*".equals(filter)) {
```

Comment:
shall we define a constant for "*" as it has been used in multiple places?

Reply:
Fixed and updated the PR.